geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/RedundancyLevelTestBase.java

```diff
@@ -132,380 +124,227 @@ public final void postSetUp() throws Exception {
-      }
+    final CacheClientNotifier ccn = bs.getAcceptor().getCacheClientNotifier();
 
-    } catch (Exception ex) {
-      Assert.fail("while setting verifyDispatcherIsNotAlive  ", ex);
+    await().atMost(180, SECONDS).pollInterval(1, SECONDS).until(() -> ccn.getClientProxies().size(),
+        greaterThan(0));
+
+    Iterator<CacheClientProxy> cacheClientProxyIterator = ccn.getClientProxies().iterator();
+    if (cacheClientProxyIterator.hasNext()) {
+      CacheClientProxy proxy = cacheClientProxyIterator.next();
+      assertThat(proxy._messageDispatcher.isAlive())
+          .describedAs("Dispatcher on secondary should not be alive").isFalse();
     }
   }
 
-  public static void verifyRedundantServersContain(final String server) {
-    WaitCriterion wc = new WaitCriterion() {
-      public boolean done() {
-        return pool.getRedundantNames().contains(server);
-      }
+  static void verifyRedundantServersContain(final String server) {
 
-      public String description() {
-        return "Redundant servers (" + pool.getRedundantNames() + ") does not contain " + server;
-      }
-    };
-    Wait.waitForCriterion(wc, 60 * 1000, 2000, true);
+    await()
+        .atMost(180, SECONDS).pollInterval(2, SECONDS)
+        .until(() -> pool.getRedundantNames().contains(server));
   }
 
-  public static void verifyLiveAndRedundantServers(final int liveServers,
+  /**
+   *
+   * @param connectedServers is the expected number of connected servers (0 or more)
+   * @param redundantServers is the expected number of redundant servers (0 or more)
+   */
+  static void verifyConnectedAndRedundantServers(final int connectedServers,
       final int redundantServers) {
-    WaitCriterion wc = new WaitCriterion() {
-      public boolean done() {
-        return pool.getConnectedServerCount() == liveServers
-            && pool.getRedundantNames().size() == redundantServers;
-      }
-
-      public String description() {
-        return "Expected connected server count (" + pool.getConnectedServerCount() + ") to become "
-            + liveServers + "and redundant count (" + pool.getRedundantNames().size()
-            + ") to become " + redundantServers;
-      }
-    };
-    Wait.waitForCriterion(wc, 120 * 1000, 2 * 1000, true);
-  }
 
-  public static void verifyDeadServers(int deadServers) {
-    // this is now deadcode since it is always followed by verifyLiveAndRedundant
-    // long maxWaitTime = 180000;
-    // long start = System.currentTimeMillis();
-    // while (proxy.getDeadServers().size() != deadServers) { // wait until condition is
-    // // met
-    // assertTrue("Waited over " + maxWaitTime + "for dead servers to become "
-    // + deadServers, (System.currentTimeMillis() - start) < maxWaitTime);
-    // try {
-    // Thread.yield();
-    // synchronized(delayLock) {delayLock.wait(4000);}
-    // }
-    // catch (InterruptedException ie) {
-    // fail("Interrupted while waiting ", ie);
-    // }
-    // }
+    await(
```

Comment:
Here is the important change for the primary problem. Elsewhere we cleaned up and normalized most of the timeouts to 3 minutes each, and used Awaitility.